_includes/shelter_map.html

```diff
@@ -1,125 +1,332 @@
-                              if (shelters[i].address) html += '<a class="directions" target="_blank" href="https://www.google.com/maps/dir/current+location/' + encodeURIComponent(shelters[i].address) + '">Directions</a>';
-                              if (note) html += '<div class="note">' + note + '</div>';
-                              if (petNote) html += '<div class="petNote">Pets: ' + petNote + '</div>';
-                              var popup = new mapboxgl.Popup({ offset: 25 })
-                                  .setHTML(html);
-                              shelters[i].marker = new mapboxgl.Marker(el)
-                                  .setLngLat([shelters[i].longitude, shelters[i].latitude])
-                                  .setPopup(popup)
-                                  .addTo(map);
-                          }
-                      };
-                      oReq.open('GET', 'https://hurricane-florence-api.herokuapp.com/api/v1/shelters', true);
-                      oReq.responseType = 'json';
-                      oReq.send();
-                      var filterRadios = document.getElementsByName('filter');
-                      for(var i = 0; i< filterRadios.length; i++)
-                          filterRadios[i].onclick = setFilter;
-                      var geolocate = document.getElementById('geolocate');
-                      if (!navigator.geolocation || !geolocate) {
-                          geolocate.style.display = 'none';
-                      }
-                      else {
-                          geolocate.onclick = function(e) {
-                              navigator.geolocation.getCurrentPosition(function(pos) {
-                                  map.flyTo({ center: [pos.coords.longitude, pos.coords.latitude], zoom: 11});
-                              });
-                              return false;
-                          }
-                      }
-                      var navControl = new mapboxgl.NavigationControl();
-                      map.addControl(navControl, 'top-right');
-                  </script>
 </div>
+<script>
+    window.shelters = null;
+    let sheltersShown = [];
+    let currentFilterActions = [];
+    const DateTime = luxon.DateTime;
+    const hours = (val) => 1000 * 60 * 60 * val;
+
+    mapboxgl.accessToken = '{{site.mapbox_access_token}}';
+    var map = new mapboxgl.Map({
+        container: 'map',
+        style: 'mapbox://styles/sbma44/cjm0qon9e7u8l2sqliotzrlmi',
+        center: [-78, 35.5],
+        zoom: 6.1
+    });
+    var oReq = new XMLHttpRequest();
+    oReq.onload = function (e) {
+        if (!(e && e.target && e.target.response && e.target.response.shelters))
+            return console.error('error fetching shelter JSON');
+        window.shelters = sheltersShown = e.target.response.shelters;
+        populateMarkers(sheltersShown);
+    };
+    oReq.open('GET', 'https://irma-api.herokuapp.com/api/v1/shelters', true);
```

Comment:
Putting a comment here so it's easy to find, the real API end point will be at hurricane-florence-api.herokuapp.com.